plugins/daapclient/__init__.py

```diff
@@ -244,6 +248,133 @@ def __init__(self, _exaile, _menu):
         self.browser.connect_to_signal('ItemRemove', self.remove_service)
 
 
+class DaapZeroconfInterface(GObject.GObject):
+    __gsignals__ = {
+        'connect': (GObject.SignalFlags.RUN_LAST, None, (GObject.TYPE_PYOBJECT,))
+    }
+
+    def new_share_menu_item(self, menu_name, service_name, address, port):
+        '''
+            This function is called to add a server to the connect menu.
+        '''
+
+        if not self.menu:
+            return
+
+        menu_item = _smi(
+            menu_name,
+            ['sep'],
+            menu_name,
+            callback=lambda *_x: self.clicked(service_name, address, port),
+        )
+        self.menu.add_item(menu_item)
+
+    def clear_share_menu_items(self):
+        '''
+            This function is used to clear all the menu items out of a menu.
+        '''
+
+        if not self.menu:
+            return
+
+        items_to_remove = [
+            item
+            for item in self.menu._items
+            if item.name not in ('manual', 'history', 'sep')
+        ]
+        for item in items_to_remove:
+            self.menu.remove_item(item)
+
+    def rebuild_share_menu_items(self):
+        '''
+            This function fills the menu with known servers.
+        '''
+        self.clear_share_menu_items()
+
+        show_ipv6 = settings.get_option('plugin/daapclient/ipv6', False)
+        items = []
+
+        for key, info in self.services.items():
+            # Strip the service type from fully-qualified service name
+            type_pos = info.name.rfind("." + info.type)
```

Comment:
I assume this is meant to only check the end of the name? If that's the case, you should use something like `if x.endswith(y): x = x[:-len(y)]`. ([`str.removesuffix`](https://docs.python.org/3.9/library/stdtypes.html#str.removesuffix) will finally come in Python 3.9.)

Reply:
Yes, this is supposed to strip the service type (`_daap._tcp.local.`) and its preceding period from the fully qualified name that zeroconf returns (e.g., `Media server._daap._tcp.local.`).

I've revised the commit to change the syntax as suggested.